py/runai_model_streamer/runai_model_streamer/file_streamer/file_streamer.py

```
@@ -184,8 +185,24 @@ def stream_files(
+        # A submission is all filesystem or all object storage - mixing them is rejected with
+        # UnsupportedBackendMix, and homogeneous_paths() above says the same thing on this side - so
+        # the first path decides for the whole list. An object-storage submission has no mount, no
+        # O_DIRECT and no alignment requirement, so it skips the probe entirely rather than asking a
+        # question that can only answer "nothing to measure".
+        #
+        # Asked AFTER handle_object_store() above, so these paths are the ones that will be read.
+        paths = [request.path for request in file_stream_requests]
+        object_storage = bool(paths) and (is_s3_path(paths[0]) or is_gs_path(paths[0])
+                                          or is_azure_path(paths[0]))
+
+        direct_block = (DIRECT_IO_BLOCK if object_storage or not paths
```

Ruff (0.16.3)

[error] Undefined name `DIRECT_IO_BLOCK`

(F821)